src/main/java/module/teamAnalyzer/ui/FilterPanel.java

```diff
@@ -139,39 +134,76 @@ private void jbInit() {
 			SystemManager.updateReport();
 		});
 
-		downloadButton.addActionListener(new ActionListener() {
-			@Override
-			public void actionPerformed(ActionEvent e) {
+		downloadButton.addActionListener(e -> {
+            downloadButton.setEnabled(false);
+			// Trigger event in a separate thread to avoid Button UI from being blocked.
+			SwingUtilities.invokeLater(() -> {
+				final ExecutorService downloadExecutor = Executors.newCachedThreadPool();
+
 				HOLogger.instance().log(getClass(),
 						"UPDATE for Team " + SystemManager.getActiveTeamId());
-				//HattrickManager.downloadPlayers(SystemManager.getActiveTeamId());
+
 				// Load squad info of all teams
-				for ( var team  : TeamManager.getTeams()){
-					HattrickManager.downloadPlayers(team.getTeamId());
+				try {
+					for (var team : TeamManager.getTeams()) {
+						Map<String, String> teamDetails = HattrickManager.getTeamDetails(team.getTeamId());
+						System.out.println(teamDetails);
```

Comment:
@tychobrailleur should be removed too?

Reply:
Well that's embarrassing. :-)  Thanks for noticing, now fixed.